src/constants/operators.ts

```diff
@@ -1,3 +1,20 @@
+export type Operator =
+    | 'NOT_IN'
+    | 'IN'
+    | 'STR_ENDS_WITH'
+    | 'STR_STARTS_WITH'
+    | 'STR_CONTAINS'
+    | 'NUM_EQ'
+    | 'NUM_GT'
+    | 'NUM_GTE'
+    | 'NUM_LT'
+    | 'NUM_LTE'
+    | 'DATE_AFTER'
+    | 'DATE_BEFORE'
+    | 'SEMVER_EQ'
+    | 'SEMVER_GT'
+    | 'SEMVER_LT';
+
 export const NOT_IN = 'NOT_IN';
```

Comment:
Now that we have the `Operator` type, we could skip the `const`s and use strings directly (since they have to match):

```
export const allOperators: Operator[] = [
    'NOT_IN',
    // ...
]
```

Reply:
Same for `Input` and `Validator` above, no `const`s required.